### ISSUE TITLE
json_parse filter function omits asset handles

### ISSUE THREAD

**Issue by Alphish** [YoYoGames/GameMaker-Bugs#4463]
### Description
When using `json_parse` with a filter function, the filter function doesn't process asset handle strings (such as "@ref room(rm_Grassland)"), instead skipping to the next item to process.

E.g. when I parsed JSON string like this:
`{"x":160.0,"y":512.0,"room":"@ref room(rm_Grassland)"}`

With the following filter function:
```gml
var _loaded = json_parse(_json, function(_key, _value) {
    show_debug_message($"Key: {_key}, Type: {typeof(_value)}, Value: {_value}");
    return _value;
});
```

I got the following output:
```
Key: x, Type: number, Value: 160
Key: y, Type: number, Value: 512
Key: , Type: struct, Value: { x : 160, y : 512, room : ref room 0 }
```

Note how there's no output entry for parsing "room" variable!

### Steps To Reproduce
1. json_stringify a struct containing an asset ref and other values, e.g. `{ x : 160, y: 512, room: rm_Grassland }` (or whatever other room there is in the project).
2. json_parse the resulting JSON, with a filter function showing debug message for key and value and returning the original value.
3. Notice how the output is missing entries for asset-containing variables.

[69c33849-50ae-4c16-8019-10d5e61df0f6](https://api.gamemaker.io/api/github/downloads/69c33849-50ae-4c16-8019-10d5e61df0f6)

**Comment by rwkay** [comment]
Fixed in 2024.2

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.200.0.478  Runtime v2024.200.0.493
